lib/gc/Analysis/MatmulConfigAnalysis.cpp

```diff
@@ -47,6 +47,12 @@ bool validateConfig(const MatmulConfig &cfg) {
       cfg.NBlock % cfg.innerMostNBlock != 0 ||
       cfg.KBlock % cfg.innerMostKBlock != 0)
     return false;
+  if (!shape.empty()) {
```

Comment:
If this is the case that generated by heuristic rule, we also need to update the logic there?
BTW, seems it only use the `K` dim size, no need to pass the entire `shape` here?

Reply:
> If this is the case that generated by heuristic rule, we also need to update the logic there? BTW, seems it only use the `K` dim size, no need to pass the entire `shape` here?

Heuristic rule will not generate this kind of invalid config. This issue is exposed by tuner side generated config. 
The reason for only checking `K` dim size is because, only the `KThreads` value will not shrink automatically while `M` and `N` would (which is an upstream limitation). I currently passed the entire `shape` here in case we further need to check shape related constraint for `M` and `N` in the future.

Reply:
I see. I would suggest we only pass the things that is necessary as you mentioned on `M` and `N`, anyway you can keep it as is.